lib/grape/dsl/routing.rb

```diff
@@ -120,7 +120,8 @@ def route(methods, paths = ['/'], route_options = {}, &block)
             }).deep_merge(route_setting(:description) || {}).deep_merge(route_options || {})
           }
 
-          endpoints << Grape::Endpoint.new(inheritable_setting, endpoint_options, &block)
+          new_endpoint = Grape::Endpoint.new(inheritable_setting, endpoint_options, &block)
+          endpoints << new_endpoint unless endpoints.any?{ |o| o.options == new_endpoint.options && o.inheritable_setting.to_hash == new_endpoint.inheritable_setting.to_hash }
```

Comment:
I would want a comparison operator on `Grape::Endpoint` so we could maybe write `endpoints.include?(...)` or `endpoints.any? { |e| endpoint.equals?(e) } # or == `.